app/models/queue_item.rb

```diff
@@ -1,10 +1,38 @@
 class QueueItem < ActiveRecord::Base
+  validate :position_cannot_be_less_than_one
+  validate :position_cannot_be_greater_than_queue_items_count
+
   belongs_to :user
   belongs_to :video
 
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  def self.save_and_update_positions(queue_ids_with_positions)
+    begin
+      ActiveRecord::Base.transaction do
+        queue_ids_with_positions.each do |id, position|
+            queue_item = QueueItem.find(id.to_i)
+            queue_item.update!(position: position)
+        end
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      return e.message
```

Comment:
Well I think `return` keyword is definitely not needed here.

HOMEWORK: Study about Ruby's 'return' more.